lib/zold/hungry_wallets.rb

```diff
@@ -19,16 +19,25 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
-
 require 'delegate'
-
+require_relative './commands/pull'
+require_relative './commands/fetch'
+require_relative 'wallet'
 module Zold
   # Wallets decorator that adds missing wallets to the queue to be pulled later.
   class HungryWallets < SimpleDelegator
-    # @todo #280:30min Add to the queue. Once in there, try
-    #  to pull it as soon as possible as is described in #280.
+    def initialize(wallets, remotes, copies)
+      @wallets = wallets
+      @remotes = remotes
+      @copies  = copies
+      super(@wallets)
+    end
+
     def find(id)
-      yield super(id)
+      Zold::Pull.new(wallets: @wallets, remotes: @remotes, copies: @copies).run(['pull', id.to_s, '--quiet-if-absent']) unless File.exist?(File.join(@wallets.path, id.to_s + Wallet::EXT))
```

Comment:
@lastk Also in this case, checking if the wallet exist outside of the `wallets` class is not OOP. It would be better to call a `wallets` method for it.

Reply:
It is a decorated wallets class. so I think it still makes sense. or not? but now I think I should at least move it for a private method inside this class.

Reply:
@lastk Maybe you can move the path of the wallet to a method, then calling it both from here and from the test. This way, if something change in the way we store the wallets, there is only one point to be changed